pypeit/core/telluric.py

```diff
@@ -2380,8 +2542,12 @@ def run(self, only_orders=None):
                                               inmask=self.mask_arr[self.ind_lower[iord]:self.ind_upper[iord]+1,iord],
                                               maxiter=self.maxiter, lower=self.lower,
                                               upper=self.upper, sticky=self.sticky)
-            self.theta_obj_list[iord] = self.result_list[iord].x[:-7]
-            self.theta_tell_list[iord] = self.result_list[iord].x[-7:]
+            if self.teltype == 'PCA':
+                self.theta_obj_list[iord] = self.result_list[iord].x[:-(self.tell_npca+3)]
+                self.theta_tell_list[iord] = self.result_list[iord].x[-(self.tell_npca+3):]
+            elif self.teltype == 'grid':
+                self.theta_obj_list[iord] = self.result_list[iord].x[:-(4+3)]
+                self.theta_tell_list[iord] = self.result_list[iord].x[-(4+3):]
```

Comment:
Perhaps it's not needed, but you might consider adding the `else` statement as well, including an error message similar to that described above.

Reply:
I've removed the else statements here (and in subsequent comments) in favor of the `validate()` routines in pypeitpar